qiskit/transpiler/runningpassmanager.py

```diff
@@ -107,11 +108,40 @@ def _to_target(self, passmanager_ir: DAGCircuit, in_program: QuantumCircuit) ->
         circuit = dag_to_circuit(passmanager_ir, copy_operations=False)
         circuit.name = self.metadata["output_name"]
 
-        if self.property_set["layout"] is not None:
+        layout = self.property_set["layout"]
+        # If  there is no initial layout set but "original_layout" is set that means
+        # `ElidePermutations` ran and we have a final layout. Since a final layout
+        # needs an initial layout to be useful we take the original layout field which
+        # is just a trivial layout at the time ElidePermutations ran (which must be
+        # before an initial layout gets set).
+        if not layout:
+            layout = self.property_set["original_layout"]
+        final_layout = self.property_set["final_layout"]
+        if not final_layout:
+            # If no initial layout is set reverse the elision final layout so it
+            # maatches expectations. If we have an initial layout set then ApplyLayout
+            # handles mapping the layout to the set initial layout and nothing needs to
+            # be done.
+            if (
+                not self.property_set["layout"]
+                and self.property_set["elision_final_layout"] is not None
```

Comment:
Is there perhaps a more unified way we can encapsulate the outputs of `ElidePermutations`? For example, storing a struct in `property_set` under a key like `"elision"`? It feels like we have to do too much deduction on entries in the `property_set` to figure out what happened, and it's a bit tough to follow.